src/main/java/org/wildfly/security/http/HttpServerScopes.java

```diff
@@ -27,6 +27,22 @@
 public interface HttpServerScopes {
 
     /**
+     * Returns {@code true} if the specified scope exists.
+     *
+     * @param scope the scope required.
+     * @return {@code true} if the specified scope exists. Otherwise, {@code false}
+     */
+    boolean exists(Scope scope);
```

Comment:
I think it will be easier to understand if these are on the scope, with this current API exists could return false and create still return null - if we get it onto the scope then having a scope means we know it is supported.

Secondly scopes that just exist and don;t support creation can always return true on exists which means create should never be called - e.g. a connection scope.